### PR TITLE
Blast download panel

### PR DESCRIPTION
**Pull request by xwatkins**
Wrap the direct BLAST download functionality into the new `SidePanel` container. Use dropdown for format selection.

### CODE REVIEW

File: src/tools/blast/components/results/BlastResultsButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const [displayDownloadPanel, setDisplayDownloadPanel] = useState(false);
+
+  return (
+    <>
+      {displayDownloadPanel && (
+        <Suspense fallback>
+          <SidePanel>
+            <BlastResultDownload
+              id={jobId}
+              onClose={() => setDisplayDownloadPanel(!displayDownloadPanel)}
```
Comment (aurel-l) [inline src/tools/blast/components/results/BlastResultsButtons.tsx:21]: Why not simply set to `false`?

Comment (xwatkins) [reply src/tools/blast/components/results/BlastResultsButtons.tsx:21]: In the first PR on side panel, @its-mahdi mentioned that the Download button can still be visible and we both thought you'd expect it to toggle the panel if you clicked on it again.

Comment (aurel-l) [reply src/tools/blast/components/results/BlastResultsButtons.tsx:21]: Well, that's definitely OK for the other place where you call `setDisplayDownloadPanel` (the download button) but here the `onClose` is quite clear about the intent of what should be done within the function.

Comment (xwatkins) [reply src/tools/blast/components/results/BlastResultsButtons.tsx]: Good point, renamed to `onToggleDisplay`

**Pull request merged**